apiml/src/main/java/org/zowe/apiml/controller/ApimlExceptionHandler.java

```diff
@@ -99,4 +100,10 @@ public Mono<Void> handleBadCredentialsException(ServerWebExchange exchange, BadC
         return setBodyResponse(exchange, SC_UNAUTHORIZED, "org.zowe.apiml.security.login.invalidCredentials");
     }
 
+    @ExceptionHandler(ZosAuthenticationException.class)
+    public Mono<Void> handleZosAuthenticationException(ServerWebExchange exchange, ZosAuthenticationException ex) {
+        log.debug("Zos Authentication Exception: {}", ex.getMessage());
+        return setBodyResponse(exchange, ex.getPlatformError().responseCode.value(), ex.getPlatformError().errorMessage, ex.getMessage());
```

Comment:
```suggestion
        return setBodyResponse(exchange, ex.getPlatformError().responseCode.value(), Optional.ofNullable(ex.getPlatformError()).map(e -> e.errorMessage).orElse(null), ex.getMessage());
```

A potential NPE (see org.zowe.apiml.security.common.error.PlatformPwdErrno)